src/Build/BackEnd/Components/SdkResolution/CachingSdkResolverService.cs

```diff
@@ -53,7 +54,8 @@ public override SdkResult ResolveSdk(int submissionId, SdkReference sdk, Logging
                  */
                 result = cached.GetOrAdd(
                     sdk.Name,
-                    key => base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive));
+                    key => new Lazy<SdkResult>(() => base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive)))
+                    .Value;
```

Comment:
I think I'd prefer a named variable for the lazy; this construction confused me for a second until I realized that there was a `)` between where I though the `.Value` was and where it actually was.

Reply:
I had it like that at first and it felt too verbose.  But the tribe has spoken...

I also added/updated the comments